lib/scrivener/paginater/ecto/query.ex

```diff
@@ -17,12 +17,20 @@ defimpl Scrivener.Paginater, for: Ecto.Query do
       Keyword.get_lazy(options, :total_entries, fn -> total_entries(query, repo, caller) end)
 
     total_pages = total_pages(total_entries, page_size)
-    page_number = min(total_pages, page_number)
+    allow_overflow_page_number = Keyword.get(options, :allow_overflow_page_number, false)
+
+    page_number =
+      if allow_overflow_page_number, do: page_number, else: min(total_pages, page_number)
+
+    items =
+      if page_number > total_pages,
+        do: [],
+        else: entries(query, repo, page_number, page_size, caller, options)
```

Comment:
Is there a reason to add this conditional? Efficiency?

Reply:
yup, no need to make an extra db query in this case

Reply:
Got it. In that case I'd prefer to inline this into the `entries` function itself, perhaps via guard clauses. It'll mean adding another argument to the method.